python_modules/libraries/dagster-components/dagster_components/lib/dbt_project.py

```diff
@@ -48,23 +46,29 @@ class DbtProjectComponentTranslator(DagsterDbtTranslator):
     def __init__(
         self,
         *,
+        value_resolver: TemplatedValueResolver,
         translator_params: Optional[DbtNodeTranslatorParams] = None,
     ):
+        self.value_resolver = value_resolver
         self.translator_params = translator_params
 
     def get_asset_key(self, dbt_resource_props: Mapping[str, Any]) -> AssetKey:
         if not self.translator_params or not self.translator_params.key:
             return super().get_asset_key(dbt_resource_props)
 
         return AssetKey.from_user_string(
-            Template(self.translator_params.key).render(node=dbt_resource_props)
+            self.value_resolver.with_context(node=dbt_resource_props).resolve(
+                self.translator_params.key
+            )
         )
 
     def get_group_name(self, dbt_resource_props) -> Optional[str]:
         if not self.translator_params or not self.translator_params.group:
             return super().get_group_name(dbt_resource_props)
 
-        return Template(self.translator_params.group).render(node=dbt_resource_props)
+        return self.value_resolver.with_context(node=dbt_resource_props).resolve(
+            self.translator_params.group
+        )
```

Comment:
Hmmm it seems problematic that with_content has to be called in two separate spots like this. This might be particular to the `DagsterDbtTranslator`, but as broad concept when we write one of these integrations there should be a canonical spot in code where we declarate what variable are in context in the document tree with metadata etc about them for debugging. 

This is very unstructured and liable to cause problems.